control-plane/pkg/reconciler/broker/broker.go

```diff
@@ -226,7 +235,34 @@ func (r *Reconciler) reconcileKind(ctx context.Context, broker *eventing.Broker)
 	}
 
 	ingressHost := network.GetServiceHostname(r.Env.IngressName, r.DataPlaneNamespace)
-	address := receiver.Address(ingressHost, broker)
+
+	transportEncryptionFlags := feature.FromContext(ctx)
+	if transportEncryptionFlags.IsPermissiveTransportEncryption() {
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpAddress := receiver.HTTPAddress(ingressHost, broker)
+		httpsAddress := receiver.HTTPSAddress(ingressHost, broker, caCerts)
+		broker.Status.Address = &httpAddress
+		broker.Status.Addresses = []pkgduckv1.Addressable{httpAddress, httpsAddress}
+	} else if transportEncryptionFlags.IsStrictTransportEncryption() {
+		caCerts, err := r.getCaCerts()
+		if err != nil {
+			return err
+		}
+
+		httpsAddress := receiver.HTTPSAddress(ingressHost, broker, caCerts)
+		broker.Status.Address = &httpsAddress
+		broker.Status.Addresses = []pkgduckv1.Addressable{httpsAddress}
+	} else {
+		httpAddress := receiver.HTTPAddress(ingressHost, broker)
+		broker.Status.Address = &httpAddress
+		broker.Status.Addresses = []pkgduckv1.Addressable{httpAddress}
+	}
+
+	address := broker.Status.Address.URL.URL()
```

Comment:
As a follow up, we need to make the prober TLS aware, because because currently if I'd enable strict transport encryption: line 265 would use the HTTPS endpoint but the prober is not configured or aware of the CA certs and it would fail

Reply:
I made an issue for this here: https://github.com/knative-sandbox/eventing-kafka-broker/issues/3128